lib/Models/ItemSearchProviders/NumericIndex.ts

```diff
@@ -73,8 +73,12 @@ export default class NumericIndex implements IndexBase<NumericSearchQuery> {
       { value: endValue },
       (a, b) => a.value - b.value
     );
-    const startIndex = i < 0 ? ~i : i;
-    const endIndex = j < 0 ? ~j : j;
+    let startIndex = i < 0 ? ~i : i;
+    let endIndex = j < 0 ? ~j : j;
+    // iterate startIndex backward till we find the first matching value
+    while (idValuePairs[startIndex - 1]?.value === value.start) startIndex -= 1;
+    // iterate endIndex forward till we find the last matching value
+    while (idValuePairs[endIndex + 1]?.value === value.end) endIndex += 1;
```

Comment:
Basically  the binary search could stop somewhere in the middle of a sequence of repeated values. 

Eg, Let's say we are searching for the value `10` in the sequence `7, 8, 9, 10, 10, 10, 10, 10, 11, 12, 13..`. `binarySearch` might return index value `6` for start & end index. In that case we have to walk backwards & forwards in the array to discover the first & last matching value, i.e 3 & 7.

Reply:
It seems a shame to do a binary search and then iterate linearly to find the final solution. Maybe you could switch out the `binarySearch` for [_.sortedIndexOf](https://lodash.com/docs/4.17.15#sortedIndexOf) and [_.sortedLastIndexOf](https://lodash.com/docs/4.17.15#sortedLastIndexOf).

Reply:
Is it possible that most of this list is the same value? Like 10,000 entries with the same value (and some that are different, or none, or whatever would make this solution act worst).